src/main/java/decodex/parser/Parser.java

```diff
@@ -174,13 +176,35 @@ private InputCommand craftInputCommand(String userInput) throws ParserException
         return new InputCommand(inputData);
     }
 
+    /**
+     * Returns the list category for the list command.
+     *
+     * @param userInput The user input specified by the user.
+     * @return The list category String.
+     * @throws CommandException If the command has more than 1 argument.
+     */
+    private String getListCategory(String userInput) throws CommandException {
+        String[] arguments = userInput.split(SPLIT_REGEX);
+        assert arguments.length > 0 : "userInput should have command word";
+
+        if (arguments.length == 1) {
+            return null;
+        } else if (arguments.length == 2) {
+            return arguments[STARTING_ARGUMENTS_INDEX];
+        } else {
+            throw new CommandException(ErrorMessages.TOO_MANY_COMMAND_ARGUMENTS);
+        }
```

Comment:
Magic numbers 1 and 2. 

Also, using the word `arguments` might be wrong because your first token, your command, is clearly not an argument. Perhaps you want to separate the argument tokens out first?